exonum-java-binding/core/rust/src/storage/db.rs

```diff
@@ -207,6 +246,25 @@ mod tests {
         let view = View::from_owned_fork(fork);
         check_fork(view.get());
         assert!(view.can_convert_into_fork());
+        assert!(view.can_rollback());
+    }
+
+    #[test]
+    fn rollback() {
+        let db = setup_database();
+        let mut fork = db.fork();
+        let mut view = View::from_ref_mut_fork(&mut fork);
+        view.create_checkpoint();
+        {
+            let mut index = match view.get() {
+                ViewRef::Fork(fork) => entry(fork),
+                _ => unreachable!("Invalid variant of ViewRef, expected Fork"),
+            };
+            index.set(SECOND_TEST_VALUE);
+        }
+        check_value(&view.get(), SECOND_TEST_VALUE);
```

Comment:
I'd consider adding some comments on the intention of this test.

Reply:
Done, also refactored it by removing code duplication